bin/publisher/manifest.py

```diff
@@ -1,76 +1,142 @@
-    return actions
+def generate_manifest(gallery_actions_folder: str, base_url: str) -> Manifest:
+    """
+    Generates the manifest file for the built action packages.
 
-def generate_manifest(gallery_actions_folder, base_url):
-    manifest = {'action_packages': []}
+    Parameters:
+        gallery_actions_folder (str): The path to the folder containing prepared gallery action packages.
+        base_url (str): The base URL for the gallery actions in the S3.
+    """
+    manifest: Manifest = { 'packages': {} }
     all_package_hashes = []
 
-    for action_name in os.listdir(gallery_actions_folder):
-        action_path = os.path.join(gallery_actions_folder, action_name)
-        if os.path.isdir(action_path):
+    for action_package_name in os.listdir(gallery_actions_folder):
+        action_package_path = os.path.join(gallery_actions_folder, action_package_name)
+
+        if os.path.isdir(action_package_path):
             versions_info = []
-            for version_dir in os.listdir(action_path):
-                version_path = os.path.join(action_path, version_dir)
+
+            for version_dir in os.listdir(action_package_path):
+                version_path = os.path.join(action_package_path, version_dir)
                 if os.path.isdir(version_path):
                     metadata_path = os.path.join(version_path, "metadata.json")
                     package_yaml_path = os.path.join(version_path, "package.yaml")
                     env_hash_path = os.path.join(version_path, "env.hash")
                     package_hash_path = os.path.join(version_path, "package.hash")
 
-                    package_data = read_package_yaml(package_yaml_path)
+                    package_data = read_yaml_file(package_yaml_path)
                     actions = read_metadata_json(metadata_path)
                     python_env_hash = read_file_contents(env_hash_path)
                     zip_hash = read_file_contents(package_hash_path)
 
-                    version_info = {
+                    version_info: VersionInfo = {
                         'version': package_data.get('version', version_dir),
                         'description': package_data.get('description', 'No description provided.'),
-                        'zip': f"{base_url}{action_name}/{version_dir}/{action_name}.zip",
-                        'icon': f"{base_url}{action_name}/{version_dir}/package.png",
-                        'metadata': f"{base_url}{action_name}/{version_dir}/metadata.json",
+                        'zip': f"{base_url}{action_package_name}/{version_dir}/{action_package_name}.zip",
+                        'icon': f"{base_url}{action_package_name}/{version_dir}/package.png",
+                        'metadata': f"{base_url}{action_package_name}/{version_dir}/metadata.json",
                         'actions': actions,
                         'python_env_hash': python_env_hash,
                         'zip_hash': zip_hash
                     }
```

Comment:
Message from @tonnitommi https://sema4ai.slack.com/archives/C012AULG4N8/p1720444500461179?thread_ts=1720437225.159449&cid=C012AULG4N8
... that we need to include the Changelog & Readme.md files into the upload. 
That means we need to upload the files but also make them available to be consumed for each version

Reply:
@custompointofview do we have some agreed conventions for the name casing? Like `changelog.md` vs `CHANGELOG.md` ?

Reply:
Also - Action Server is the actual entity building the packages. I assume we should include readme and changelog files there already right?

Reply:
Nvm, it seems like Action Server bundles `CHANGELOG.md` anyway ^^

Reply:
@custompointofview done!